.github/workflows/pull-request.yml

```diff
@@ -123,6 +123,95 @@ jobs:
           path: './coverage/lcov.info'
           min_coverage: 95.0
 
+  Health-Check:
+    name: Server health check
+    runs-on: ubuntu-latest
+    needs: Test-Application
+    strategy:
+      matrix:
+        node-version: [14.x]
+    services:
+      mongo:
+        image: mongo:4.4
+        options: >-
+          --health-cmd mongo
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 10
+        ports:
+          - 27017:27017
+    env:
+      MONGO_DB_URL: "mongodb://localhost:27017/talawa"
+      RECAPTCHA_SITE_KEY: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+      RECAPTCHA_SECRET_KEY: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+      MAIL_USERNAME: "abc@gmail.com"
+      MAIL_PASSWORD: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+      IS_SMTP: ""
+      SMTP_HOST: ""
+      SMTP_PASSWORD: ""
+      SMTP_USERNAME: ""
+      SMTP_PORT: ""
+      SMTP_SSL_TLS: ""
+      apiKey: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+      appId: "1:XXXXXXXXXXXX:android:XXXXXXXXXXXXXXXXXXXXXX"
+      messagingSenderId: "XXXXXXXXXXXX"
+      projectId: "appname-XXXXXX"
+      storageBucket: "appname-XXXXXX.appspot.com"
+      iOSapiKey: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+      iOSappId: "1:XXXXXXXXXXXX:ios:XXXXXXXXXXXXXXXXXXXXXX"
+      iOSmessagingSenderId: "XXXXXXXXXXXX"
+      iOSprojectId: "appame-XXXXXX"
+      iOSstorageBucket: "appame-XXXXXX.appspot.com"
+      iosClientId: "XXXXXXXXXXXX-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX.apps.googleusercontent.com"
+      iosBundleId: "com.appname.app"
+      LAST_RESORT_SUPERADMIN_EMAIL: "abc@gmail.com"
+      COLORIZE_LOGS: "true"
+      LOG_LEVEL: "info"
+
+    steps:
+      - name: Checkout 
+        uses: actions/checkout@v3
+
+      - name: Generate Access Token Secret
+        run: echo "ACCESS_TOKEN_SECRET=$(openssl rand -hex 32)" >> $GITHUB_ENV
+
+      - name: Generate Refresh Token Secret
+        run: echo "REFRESH_TOKEN_SECRET=$(openssl rand -hex 32)" >> $GITHUB_ENV
+
+      - name: Set up Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Start the development server 
+        run: |
+          npm run dev &
+          echo "Development server started..."
+      # Sleep for 10 s
+      - name: Sleep for 10s
+        uses: juliangruber/sleep-action@v1
+        with:
+          time: 10s
+
+      #check http server status
+      - name: Check server status
```

Comment:
It may be better for us to start the API after the `npm run test` section so that all the pre-requisite steps are not required as the testing workflow would have installed them already. 

It would be much simpler and starting the API after testing would still occur as was the intention of the original issue.

Reply:
that sir, was the first approach that struck my mind and i merged this in the test job itself which wouldn't cause us to load the mongodb or the checkout twice


![image](https://github.com/PalisadoesFoundation/talawa-api/assets/117587128/8872424d-2744-406b-a4ea-e2a2be3f8c75)

but one of our reviewers advised me to make it a separate job as the test job didn't need it


![image](https://github.com/PalisadoesFoundation/talawa-api/assets/117587128/2cedb0b1-e256-4027-9ab2-20c97085418c)
*apologies if i misunderstood something*

Reply:
What was said is true, the tests don't need a running server.

But we need to make sure the API starts after all the tests are done to be safe. After testing, not before.

If that is achieved then we're OK. The duplication of installation is going to slow down everything